TopPlaces/TopPlaces/Model/PlacesPhotosProvider.h

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2016 Lightricks. All rights reserved.
+// Created by Gennadi Iosad.
+
+NS_ASSUME_NONNULL_BEGIN
+
+/// Abstraction for a place description that can be presented to the user
+@protocol PlaceInfo<NSObject>
+
+/// Returns the URL to download the list of the photos from the place,
+/// the list length is bounded by \c maxLength
+- (NSURL*)urlOfPhotoInfoArrayWithMaxLength:(NSUInteger)maxLength;
+
+/// Compares to another instance of PlaceInfo conforming object
+- (NSComparisonResult)compare:(id<PlaceInfo>)other;
+
+/// Returns the title of the place
+@property (nonatomic, readonly) NSString *title;
+
+/// Returns additional details of the place
+@property (nonatomic, readonly) NSString *details;
+
+/// Returns the country of the place
+@property (nonatomic, readonly) NSString *country;
+
+@end
+
+
+/// Photo information absraction
+@protocol PhotoInfo<NSCoding>
+
+/// Returns the title of the image
+@property (nonatomic, readonly) NSString *title;
+
+/// Returns the details - auxillary info of the image
+@property (nonatomic, readonly) NSString *details;
+
+/// Returns the url for downloading the image
+@property (nonatomic, readonly) NSURL *url;
+
+@end
+
+
+//Cancellable download task returned by PlacesPhotosProvider
+@protocol CancellableTask<NSObject>
+
+/// cancel download in progress, completion handler will be called with error
+- (void) cancel;
+
+@end
+
+
+/// Places and image lists provider abtraction
+@protocol PlacesPhotosProvider
+
+typedef void (^DownloadPlacesCompleteBlock)(NSArray<id<PlaceInfo>> *placesInfo, NSError *error);
```

Comment:
1) Please add short description of the purpose and parameters of this callback.
2) Note that both placesInfo and error should marked as nullable (if the download succeeds error will be nil otherwise the placesInfo will be nil).

This relevant for DownloadPhotosInfoCompleteBlock & DownloadPhotoCompleteBlock.